iceoryx_hoofs/include/iceoryx_hoofs/internal/log/platform_building_blocks/logger.inl

```diff
@@ -0,0 +1,137 @@
+}
+
+
+template <typename BaseLogger>
+inline constexpr LogLevel Logger<BaseLogger>::minimalLogLevel() noexcept
+{
+    return Logger::MINIMAL_LOG_LEVEL;
+}
+
+template <typename BaseLogger>
+inline constexpr bool Logger<BaseLogger>::ignoreLogLevel() noexcept
+{
+    return Logger::IGNORE_ACTIVE_LOG_LEVEL;
+}
+
+
+template <typename BaseLogger>
+inline Logger<BaseLogger>& Logger<BaseLogger>::get() noexcept
+{
+    // NOLINTJUSTIFICATION needed for the functionality
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+    thread_local static Logger* logger = Logger::activeLogger();
+    if (!logger->m_isActive.load(std::memory_order_relaxed))
+    {
+        // no need to loop until m_isActive is true since this is an inherent race
+        //   - the logger needs to be active for the whole lifetime of the application anyway
+        //   - if the logger was changed again, the next call will update the logger
+        //   - furthermore, it is not recommended to change the logger more than once
+        logger = Logger::activeLogger();
+    }
+    return *logger;
+}
+
+template <typename BaseLogger>
+inline void Logger<BaseLogger>::init(const LogLevel logLevel) noexcept
+{
+    Logger::get().initLoggerInternal(logLevel);
+}
+
+template <typename BaseLogger>
+inline void Logger<BaseLogger>::setActiveLogger(Logger<BaseLogger>* newLogger) noexcept
+{
+    Logger::activeLogger(newLogger);
+}
+
+template <typename BaseLogger>
+inline Logger<BaseLogger>* Logger<BaseLogger>::activeLogger(Logger<BaseLogger>* newLogger) noexcept
+{
+    static std::mutex mtx;
+    std::lock_guard<std::mutex> lock(mtx);
+    static Logger defaultLogger;
+
+    // NOLINTJUSTIFICATION needed for the functionality
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+    static Logger* logger{&defaultLogger};
+
+    if (newLogger)
+    {
+        if (logger->m_isFinalized.load(std::memory_order_relaxed))
+        {
+            logger->createLogMessageHeader(__FILE__, __LINE__, __FUNCTION__, LogLevel::ERROR);
+            logger->logString("Trying to replace logger after already initialized!");
+            logger->flush();
+            newLogger->createLogMessageHeader(__FILE__, __LINE__, __FUNCTION__, LogLevel::ERROR);
+            logger->logString("Trying to replace logger after already initialized!");
+            newLogger->flush();
+            /// @todo iox-#1345 call error handler after the error handler refactoring was merged
+        }
+        else
+        {
+            logger->m_isActive.store(false);
+            logger = newLogger;
+        }
+    }
+
+    return logger;
+}
+
+template <typename BaseLogger>
+inline void Logger<BaseLogger>::initLoggerInternal(const LogLevel logLevel) noexcept
+{
+    if (!m_isFinalized.load(std::memory_order_relaxed))
+    {
+        BaseLogger::setLogLevel(logLevel);
+        BaseLogger::initLogger(logLevel);
+        m_isFinalized.store(true, std::memory_order_relaxed);
```

Comment:
OK, we can only set it once (from default) and then never change it. If we like to, we could have a separate method to `finalize` the logger and call it explicitly. For the current use case it is not needed though.

Reply:
@MatthiasKillat yes, for the MVP it will only be possible to set it once. Eventually there will be a `finalize` function but currently it is sufficient to set it once and to be honest I also cannot think of a use case to change it multiple times.

Reply:
@MatthiasKillat @elfenpiff after some thinking I'm again more in favor of implicitly finalize the exchange of the logger on the `init` call. I cannot think of any use case where it would be useful to change the logger multiple times. If it is not done in `init`, one needs to do an additional call which can be forgotten. Nevertheless, if you disagree, lets continue this discussion in the design document in order to not block this PR